src/k8s-extension/azext_k8s_extension/partner_extensions/azuremonitormetrics/azuremonitorprofile.py

```diff
@@ -0,0 +1,87 @@
+
+
+# pylint: disable=line-too-long
+def link_azure_monitor_profile_artifacts(
+        cmd,
+        cluster_rp,
+        cluster_subscription,
+        cluster_resource_group_name,
+        cluster_name,
+        configuration_settings,
+        cluster_type
+):
+    cluster_region = get_cluster_region(cmd, cluster_rp, cluster_subscription, cluster_resource_group_name, cluster_name, cluster_type)
+    # MAC creation if required
+    azure_monitor_workspace_resource_id, azure_monitor_workspace_location = get_azure_monitor_workspace_resource(cmd, cluster_subscription, cluster_region, configuration_settings)
+    # DCE creation
+    dce_resource_id = create_dce(cmd, cluster_subscription, cluster_resource_group_name, cluster_name, azure_monitor_workspace_location)
+    # DCR creation
+    dcr_resource_id = create_dcr(cmd, azure_monitor_workspace_location, azure_monitor_workspace_resource_id, cluster_subscription, cluster_resource_group_name, cluster_name, dce_resource_id)
+    # DCRA creation
+    create_dcra(cmd, cluster_region, cluster_subscription, cluster_resource_group_name, cluster_name, dcr_resource_id)
+    # Link grafana
+    link_grafana_instance(cmd, azure_monitor_workspace_resource_id, configuration_settings)
+    # create recording rules and alerts
+    create_rules(cmd, cluster_subscription, cluster_resource_group_name, cluster_name, azure_monitor_workspace_resource_id, azure_monitor_workspace_location, configuration_settings)
+
+
+# pylint: disable=line-too-long
+def unlink_azure_monitor_profile_artifacts(cmd, cluster_subscription, cluster_resource_group_name, cluster_name):
+    # Remove DC* if prometheus is enabled
+    dc_objects_list = get_dc_objects_list(cmd, cluster_subscription, cluster_resource_group_name, cluster_name)
+    delete_dc_objects_if_prometheus_enabled(cmd, dc_objects_list, cluster_subscription, cluster_resource_group_name, cluster_name)
+    # Delete rules (Conflict({"error":{"code":"InvalidResourceLocation","message":"The resource 'NodeRecordingRulesRuleGroup-<clustername>' already exists in location 'eastus2' in resource group '<clustername>'.
+    # A resource with the same name cannot be created in location 'eastus'. Please select a new resource name."}})
+    delete_rules(cmd, cluster_subscription, cluster_resource_group_name, cluster_name)
+
+
+# pylint: disable=too-many-locals,too-many-branches,too-many-statements,line-too-long
+def ensure_azure_monitor_profile_prerequisites(
+        cmd,
+        cluster_rp,
+        cluster_subscription,
+        cluster_resource_group_name,
+        cluster_name,
+        configuration_settings,
+        cluster_type
+):
+    cloud_name = cmd.cli_ctx.cloud.name
+    if cloud_name.lower() == 'azurechinacloud':
+        raise CLIError("Azure China Cloud is not supported for the Azure Monitor Metrics extension")
+
+    if cloud_name.lower() == "azureusgovernment":
+        if configuration_settings is not None and 'grafana-resource-id' in configuration_settings:
```

Comment:
>'grafana-resource-id' [](http://example.com/codeflow?start=50&length=21)

Is this case sensitive check?  IOW, if this key was input in different case, will this still work?